pkg/resource/deploy/import.go

```diff
@@ -194,7 +194,8 @@ func (i *importer) registerProviders(ctx context.Context) (map[resource.URN]stri
 			ref := string(imp.Provider)
 			if state, ok := i.deployment.olds[imp.Provider]; ok {
 				r, err := providers.NewReference(imp.Provider, state.ID)
-				contract.AssertNoError(err)
+				contract.AssertNoErrorf(err,
+					"could not create provider reference with URN %q and ID %q", imp.Provider, state.ID)
```

Comment:
Why is this %q but everywhere before was %v?

Reply:
%q automatically quotes the string. that's generally better for error messages; makes the variable stand out:

    reference with URN foo and [..]
    reference with URN "foo" and [..]

And if the string is invalid because it's empty, it's even better than %v:

    reference with URN  and [..]
    reference with URN "" and [..]